rayon-core/src/registry.rs

```diff
@@ -191,15 +190,15 @@ where
     let mut result = Err(ThreadPoolBuildError::new(
         ErrorKind::GlobalPoolAlreadyInitialized,
     ));
+
     THE_REGISTRY_SET.call_once(|| {
-        result = registry().map(|registry| {
-            let registry = leak(registry);
-            unsafe {
+        result = registry()
+            .map(|registry: Arc<Registry>| unsafe {
                 THE_REGISTRY = Some(registry);
-            }
-            registry
-        });
+            })
+            .map(|_| unsafe { THE_REGISTRY.as_ref().unwrap() })
```

Comment:
I think we can fold these two maps together using `THE_REGISTRY.get_or_insert(registry)`, knowing that it's already `None`, and then we don't need an `unwrap`.

Reply:
Indeed!